src/prerequisites/tests/test_signals.py

```diff
@@ -154,3 +156,74 @@ def test_update_cache_triggered_by_parent_object_deletion(self, task):
 
         quest.delete()  # doesn't call task because parent_object no longer exists.
         self.assertEqual(task.call_count, 1)
+
+    def test_on_quest_badge_save_with_rank_prereq__creation(self):
+        """  Test that creating a Prereq, where either the prereq_object or the or_prereq_object are a Rank,
+        results in the generation of a new CytoScape map with the Rank as the map's initial_object.
+
+        Prereq - the model (a group of conditions)
+        prereq_object - the actual prerequisite that needs to be met
+        or_prereq_object - the alternate prerequisite.
+        """
+
+        # create prereq with rank as the first prereq
+        # and check if new map is created
+        first_rank = baker.make('courses.rank', name='first rank')
+        baker.make(
+            Prereq,
+            parent_object=baker.make('quest_manager.quest'),
+            prereq_object=first_rank,
+        )
+
+        self.assertTrue(CytoScape.objects.filter(
+            name=f'{first_rank.name} Map',
+            initial_content_type=ContentType.objects.get_for_model(first_rank),
+            initial_object_id=first_rank.id,
+        ).exists())
+
+        # create prereq with rank as the second prereq
```

Comment:
Sorry I didn't catch this the first time.  What do you mean by "second prereq"?  Do you mean it has two different prerequisites: first_rank AND second_rank?

Or did you mean an alternate OR prereq that includes the rank?  I don't see any tests here for the `or_prereq_object ` so wondering if this was supposed to be it.

Reply:
changed "second preq" to "alternate prereq", and now using `or_prereq_object` instead of `prereq_object`